frontend/src/components/ThemeProvider.vue

```diff
@@ -0,0 +1,37 @@
+<script setup lang="ts">
+import { ref, shallowRef, provide, watchEffect } from 'vue';
+import { capitalize, camelize } from 'vue';
+import { ThemeKey } from '@/components/ThemeKey';
+
+export interface Props {
+  initialTheme?: string;
```

Comment:
Pouzit provider pattern je podle me spravna cesta, zda se, ze je to "Vue native way". 

Jenom mi prijde skoda, ze ten Provider dela zbytecne moc veci a drzi si stav. To nam to cele komplikuje. Bezne by provider mel dostat `theme: string` treba `blue-example` (`<ThemeProvider :theme="blue-example">`) a pushnout pod sebe do stromu ten theme.  Nic vic, nic min.

Pushovat navic i "selectTheme", zavadet stav a koncept `initialTheme` vs `selectedTheme` to mi prijde navic a porusujici jednoduchost toho patternu.

Reply:
Kdyz uz to ma napsany tak to muzes prehodit do jinyho komponentu a ten Theme provider fakt nechat jenom uplne jednoduchej

Reply:
OK, vytaženo do wrapperu a theme provider je stateless.